File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var smallestIndex = function(nums) {

    const n = nums.length

    for(let i = 0; i < n; i++){

        let sum = 0
        let digit = nums[i]

        while(digit > 0){

            sum = sum + (digit % 10)
            digit = Math.floor(digit / 10)
        }

        if(sum == i) return i
    }

    return -1
};

File: 3550-smallest-index-with-digit-sum-equal-to-index/3550-smallest-index-with-digit-sum-equal-to-index.java
class Solution {
    public int smallestIndex(int[] nums) {
        int n = nums.length;

        for(int i = 0; i < n; i++){

            int sum = 0; 
            int number = nums[i];

            while(number > 0){
                
                int rem = number % 10;
                sum += rem;
                number /= 10;
            }

            if( sum == i ) return i;
        }
        return -1;
    }
}